### PR TITLE
Node handling

### PR DESCRIPTION
**Pull request by avigoldman**
This PR aims to make our node handling more Gatsby-like.

Instead of transforming nodes through our own custom API, we will hook into the mediaType `text/markdown`. This will let us follow the same parent/child relationship pattern that the Gatsby remark plugin works with.

### CODE REVIEW

File: packages/gatsby-mdx/utils/create-mdx-node.js [hunk 1]
```diff
@@ -38,19 +24,19 @@ module.exports = async (
...
 
-  mdxNode.meta = meta;
   mdxNode.excerpt = frontmatter.excerpt;
   mdxNode.exports = nodeExports;
-  mdxNode.rawBody = nodeContent;
+  mdxNode.rawBody = content;
+
+  // Add path to the markdown file path
+  if (node.internal.type === `File`) {
+    mdxNode.fileAbsolutePath = node.absolutePath;
```
Comment (ChristopherBiscardi) [inline packages/gatsby-mdx/utils/create-mdx-node.js]: seems like a `createNodeField` would be better for the file path, wdyt? I recognize that this is probably for API compatibility with transformer-remark but it feels weird to have in the main schema when it's already accessible via the parent node and only accessible for files anyway.

File: packages/gatsby-mdx/utils/default-options.js [hunk 2]
```diff
@@ -10,26 +10,24 @@ module.exports = pluginOptions => {
     {
       defaultLayouts: {},
       extensions: [".mdx"],
+      handleMarkdown: true,
```
[the remark this answers is not in the document]
Comment (avigoldman) [reply packages/gatsby-mdx/utils/default-options.js]: I'm fine not having this option, and just doing the try/catch.

File: packages/gatsby-mdx/utils/create-mdx-node.js (hunk 1 above)
Comment (avigoldman) [reply packages/gatsby-mdx/utils/create-mdx-node.js]: When you're creating the node, Gatsby doesn't seem to suggest using `createNodeField` – that is for modifying existing nodes you don't control. Gatsby will throw an error if you try to set the `fields` field on a node you create. The more we can stay inline with the remark plugin, the better. I see no harm in copying this pattern.

File: examples/using-contentful/gatsby-config.js [hunk 3]
```diff
@@ -18,14 +18,6 @@ module.exports = {
       resolve: `gatsby-mdx`,
       options: {
         root: __dirname,
```
[the remark this answers is not in the document]
Comment (avigoldman) [reply examples/using-contentful/gatsby-config.js]: This gets removed in #123 when it is replaced with using the Gatsby's state instead.

File: examples/using-contentful/gatsby-node.js [hunk 4]
```diff
@@ -96,12 +68,38 @@ exports.onCreateNode = async ({ node, actions, getNode }) => {
   const { createNodeField } = actions
 
   if (node.internal.type === `Mdx`) {
-    const value = createFilePath({ node, getNode })
+    let slug, title, date
+    if (node.fileAbsolutePath) {
```
[the remark this answers is not in the document]
Comment (avigoldman) [reply examples/using-contentful/gatsby-node.js]: Good call 👍

**Comment by ChristopherBiscardi** [comment]
moving to #144

**Pull request closed without merging**